src/GeekLearning.Storage.FileSystem/FileSystemStore.cs

```diff
@@ -130,21 +199,19 @@ private void EnsurePathExists(string path)
             }
         }
 
-        private Internal.FileSystemFileReference InternalGetAsync(IPrivateFileReference file)
+        private static string GenerateEtag(string fileSystemPath)
         {
-            var reference = InternalGetOrCreateAsync(file);
-            if (File.Exists(reference.FileSystemPath))
+            var data = File.ReadAllBytes(fileSystemPath);
```

Comment:
I think we should refrain from using `ReadAllBytes` kind of API in this library internals as they might be put the memory under high pressure. We should rely on a stream and  a buffer using TransformBlock : https://msdn.microsoft.com/en-us/library/system.security.cryptography.hashalgorithm.transformblock(v=vs.110).aspx